crates/wasi-http/src/types.rs

```diff
@@ -172,8 +175,11 @@ async fn handler(
         .map_err(|_| timeout_error("connection"))??;
 
         let worker = preview2::spawn(async move {
-            conn.await?;
-            Ok::<_, types::Error>(())
+            match conn.await {
+                Ok(()) => {}
+                // TODO: same as above, shouldn't throw this error away.
+                Err(e) => tracing::warn!("dropping error {e}"),
+            }
```

Comment:
I'll note that this, and the above, are somewhat unrelated but also tangential changes since I was touching these types. The above task previously dropped the error and this task tried to forward it along. In both cases though the result of this task was never actually looked at as the handle was exclusively used to keep the task alive and that's it.

I've updated this so both tasks consistently drop the error. I don't know how to communicate this error through the WASI APIs myself.